depends: Fix cross-compiling on macOS

hebasto opened the pull request:
This PR:
1. Specifies Objective C/C++ compilers for `native_qt` package.
2. Removes the `-t` option, which is incompatible with `mv` on macOS.

Fixes https://github.com/bitcoin/bitcoin/issues/32208.

laanwj commented on depends/packages/native_qt.mk:
```diff
@@ -148,5 +150,5 @@ endef
 define $(package)_postprocess_cmds
   rm -rf doc/ && \
-  mv -t .. translations/
+  mv translations/ ..
```
Had to look this up, but apparently the only thing `-t` does is to set the target directory explicitly. So `mv translations ..` is equivalent.

hebasto committed:
depends: Specify Objective C/C++ compilers for `native_qt` package

This change fixes cross-compilation from macOS to macOS with another
architecture.

hebasto committed:
depends: Fix `mv` command compatibility with macOS

janb84 commented:
I'm trying to test this PR, but having trouble verifying it. Am I correct to conclude that the fix is really mac only ( not nix-shell on macos? or GUIX cross-compile ? ) 

Can confirm that the `mv -t` option is not supported on MacOS .

hebasto commented:
> I'm trying to test this PR, but having trouble verifying it. Am I correct to conclude that the fix is really mac only ( not nix-shell on macos? or GUIX cross-compile ? )
> 
> Can confirm that the `mv -t` option is not supported on MacOS .

This PR is specific to cases like https://github.com/bitcoin/bitcoin/issues/32208.

janb84 commented:
> > I'm trying to test this PR, but having trouble verifying it. Am I correct to conclude that the fix is really mac only ( not nix-shell on macos? or GUIX cross-compile ? )
> > Can confirm that the `mv -t` option is not supported on MacOS .
> 
> This PR is specific to cases like #32208.

Have installed homebrew and have successfully recreated the error as #32208. 
This commit will resolve the error from #32208 but I do get an different error back, not sure if related to `Specifies Objective C/C++ compilers for native_qt package.` that's  fixed in this PR (or unrelated) 

<details>
Note: Using static linking will disable the use of dynamically loaded plugins. Make sure to import all needed static plugins, or compile needed modules into the library.

WARNING: QDoc will not be compiled, probably because clang's C and C++ libraries could not be located. This means that you cannot build the Qt documentation.
You may need to set CMAKE_PREFIX_PATH or LLVM_INSTALL_DIR to the location of your llvm installation.
Other than clang's libraries, you may need to install another package, such as clang itself, to provide the ClangConfig.cmake file needed to detect your libraries. Once this
file is in place, the configure script may be able to detect your system-installed libraries without further environment variables.
On macOS, you can use Homebrew's llvm package.
You will also need to set the FEATURE_clang CMake variable to ON to re-evaluate this check.
WARNING: QDoc will not be compiled because the QmlPrivate library wasn't found.
WARNING: Clang-based lupdate parser will not be available. Suitable LLVM and Clang C++ libraries have not been found.
You will need to set the FEATURE_clangcpp CMake variable to ON to re-evaluate this check.
WARNING: Qt Assistant will not be compiled because it requires Qt Network.
WARNING: Qt Designer will not be compiled because it requires Qt Network.

ERROR: detected a std::atomic implementation that fails for function pointers.  Please apply the patch corresponding to your Standard Library vendor, found in qtbase/config.tests/atomicfptr

CMake Error at qtbase/cmake/QtBuildInformation.cmake:208 (message):
  Check the configuration messages for an error that has occurred.
Call Stack (most recent call first):
  qtbase/cmake/QtBuildInformation.cmake:39 (qt_configure_print_summary)
  qtbase/cmake/QtBaseTopLevelHelpers.cmake:84 (qt_print_feature_summary)
  qtbase/cmake/QtBaseTopLevelHelpers.cmake:72 (qt_internal_print_top_level_info)
  CMakeLists.txt:113 (qt_internal_top_level_end)


-- Configuring incomplete, errors occurred!
CMake Error at qtbase/cmake/QtProcessConfigureArgs.cmake:1077 (message):
  CMake exited with code 1.
</details>

hebasto commented:
@janb84 
> ... but I do get an different error back

Please provide more details about your build environment (OS version, architecture, compiler version etc) and exact steps to reproduce the error.

janb84 commented:
> @janb84
> 
> > ... but I do get an different error back
> 
> Please provide more details about your build environment (OS version, architecture, compiler version etc) and exact steps to reproduce the error.

Of course, here you go, ill will also try to figure it out also but please let me know if you need any other details. 
 
Build env:
MacOS 15.4.1, M1 
QT 6.9.0
Tried both Apple clang version 17.0.0 (clang-1700.0.13.3) & Homebrew clang version 20.1.3
SDK 15.4 

Steps to reproduce:
- Clean previous builds artefacts  
- execute from bitcoin folder `make -C depends -j8 HOST=x86_64-apple-darwin`

janb84 reviewed (commented):
ACK [d0cce41](https://github.com/bitcoin/bitcoin/pull/32215/commits/d0cce4172c041fc9b2910b360fe496b1102b19d2)

Recreated issue #32208 on master, confirmed it's solved after this PR.

fanquake reviewed (approved):
ACK d0cce4172c041fc9b2910b360fe496b1102b19d2
